pytorch_lightning/trainer/trainer.py

```diff
@@ -2599,10 +2599,11 @@ def logger(self) -> Optional[LightningLoggerBase]:
         if len(self.loggers) == 1:
             return self.loggers[0]
         else:
-            rank_zero_warn(
-                "Using trainer.logger when Trainer is configured to use multiple loggers."
-                " This behavior will change in v1.8 when LoggerCollection is removed, and"
-                " trainer.logger will return the first logger in trainer.loggers"
+            rank_zero_deprecation(
+                "Using `trainer.logger` when Trainer is configured to use multiple loggers."
+                " `LoggerCollection` is deprecated in v1.6 in favor of `trainer.loggers` and will be removed in v1.8."
+                " This behavior will change in v1.8 such that `trainer.logger` will return the first"
+                " logger in `trainer.loggers`."
```

Comment:
Shouldn't this be inside the `LoggerCollection.__init__`?

Reply:
It could be, this is still up for discussion. @daniellepintz and I talked about it earlier and we concluded that since there was already a warning here, it made sense to replace it with a deprecation warning. 

Most people who are using `LoggerCollection` are doing so through `trainer.logger` so they should be made aware of both the deprecation of `LoggerCollection` and the changing behavior of `trainer.logger`.  

What we could do is have one warning inside the `trainer.logger` getter and another warning in `LoggerCollection.__init__`. The `trainer.logger` getter constructs a `LoggerCollection` each time it's called so both warnings would be displayed.

Reply:
For now, having two deprecation warnings makes the most sense to me, so I went ahead and added a deprecation message to `LoggerCollection.__init__`.

Reply:
I agree it makes sense to have a warning in both places